### PR TITLE
[ADD] mail_autogenerated_header

### PR DESCRIPTION
**Pull request by hbrunn**
Autogenerated headers
=====================

This module was written to mark Odoo's emails as being autogenerated according to [RFC 3834](https://tools.ietf.org/html/rfc3834), section 5. This allows receiving mail servers to act accordingly by for example not sending a vacation autoreply.

On the receiving side, this module drops all autogenerated incoming e-mails.

The combination of both avoids possible mail loops with misconfigured or broken email servers on the opposite side.

Usage
=====

There's nothing the user has to do. Developers can set the context flag ``mail_autogenerated_header_disable`` in calls to ``send_email`` in order to suppress adding any headers at all, and override ``_message_route_drop_autoreply`` to fine tune dropping autogenerated mails per model.

Known issues / Roadmap
======================

* a configuration of this per message subtype would be nice

### CODE REVIEW

**Comment by hbrunn** [comment]
tests are missing, will add them some time soon

**Comment by pedrobaeza** [comment]
Isn't this interesting to be proposed to Odoo master?

File: mail_autogenerated_header/tests/test_mail_autogenerated_header.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openerp.tests.common import TransactionCase
+
+
+class TestMailAutogeneratedHeader(TransactionCase):
+    def test_mail_autogenerated_header(self):
+        pass
```
Comment (yajo) [inline mail_autogenerated_header/tests/test_mail_autogenerated_header.py]: Missing

**Review by yajo — DISMISSED** [review]

File: mail_autogenerated_header/tests/test_mail_autogenerated_header.py (hunk 1 above)
Comment (hbrunn) [reply mail_autogenerated_header/tests/test_mail_autogenerated_header.py]: thanks, finally with tests

**Comment by hbrunn** [comment]
@Yajo I was about to close my old PRs, but this one actually seems fine to me. Could you re-review?

**Comment by yajo** [comment]
Sorry, we don't use v10 anymore 🤷🏼‍♂️ 

Do as you please.

**Comment by pedrobaeza** [comment]
@Yajo then remove your "Requested changes" review at least.

**Pull request closed without merging**